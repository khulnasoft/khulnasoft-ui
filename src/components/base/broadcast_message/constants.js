export const TYPE_BANNER = 'banner';
export const TYPE_NOTIFICATION = 'notification';
export const TYPE_LIST = [TYPE_BANNER, TYPE_NOTIFICATION];
