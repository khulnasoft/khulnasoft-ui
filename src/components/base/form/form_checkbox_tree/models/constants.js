export const CHECKED_STATE = {
  UNCHECKED: 0,
  INDETERMINATE: 1,
  CHECKED: 2,
};

export const V_MODEL = {
  PROP: 'value',
  EVENT: 'change',
};

export const QA_PREFIX = 'form_checkbox_tree_node_';
