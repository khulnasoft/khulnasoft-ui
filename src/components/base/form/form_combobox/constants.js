export const stringTokenList = [
  'giraffe',
  'dog',
  'dodo',
  'komodo dragon',
  'hippo',
  'platypus',
  'jackalope',
  'quetzal',
  'badger',
  'vicuña',
  'whale',
  'xenarthra',
];

export const labelText = 'Animals We Tolerate';

export const objectTokenList = [
  { id: '1', title: 'giraffe' },
  { id: '2', title: 'dog' },
  { id: '3', title: 'dodo' },
  { id: '4', title: 'komodo dragon' },
  { id: '5', title: 'hippo' },
  { id: '6', title: 'platypus' },
  { id: '7', title: 'jackalope' },
  { id: '8', title: 'quetzal' },
  { id: '9', title: 'badger' },
  { id: '10', title: 'vicuña' },
  { id: '11', title: 'whale' },
  { id: '12', title: 'xenarthra' },
];

export const oneTokenList = ['dog'];

export const actionsList = [
  {
    label: 'Create',
    fn: () => {
      // eslint-disable-next-line no-alert
      window.alert('Create action');
    },
  },
  {
    label: 'Edit',
    fn: () => {
      // eslint-disable-next-line no-alert
      window.alert('Edit action');
    },
  },
];
