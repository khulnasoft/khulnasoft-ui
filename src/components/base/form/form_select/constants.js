export const formSelectOptions = [
  { value: 'Pizza', text: 'Pizza' },
  { value: 'Tacos', text: 'Tacos' },
  { value: 'Burger', text: 'Burger' },
];
