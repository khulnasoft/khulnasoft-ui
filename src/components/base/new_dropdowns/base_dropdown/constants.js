export const FIXED_WIDTH_CLASS = '!gl-w-31';
export const DEFAULT_OFFSET = 4;
