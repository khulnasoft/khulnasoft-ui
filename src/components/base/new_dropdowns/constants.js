// base dropdown events
export const GL_DROPDOWN_SHOWN = 'shown';
export const GL_DROPDOWN_HIDDEN = 'hidden';
export const GL_DROPDOWN_BEFORE_CLOSE = 'beforeClose';
export const GL_DROPDOWN_FOCUS_CONTENT = 'focusContent';

// KEY Codes
export const ARROW_DOWN = 'ArrowDown';
export const ARROW_UP = 'ArrowUp';
export const END = 'End';
export const ENTER = 'Enter';
export const HOME = 'Home';
export const SPACE = 'Space';

// Positioning strategies
// https://floating-ui.com/docs/computePosition#strategy
export const POSITION_ABSOLUTE = 'absolute';
export const POSITION_FIXED = 'fixed';

export const GL_DROPDOWN_CONTENTS_CLASS = 'gl-new-dropdown-contents';
