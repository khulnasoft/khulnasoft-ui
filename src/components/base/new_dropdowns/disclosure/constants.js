export const DISCLOSURE_DROPDOWN_ITEM_NAME = 'GlDisclosureDropdownItem';
export const DISCLOSURE_DROPDOWN_GROUP_NAME = 'GlDisclosureDropdownGroup';
export const DISCLOSURE_DROPDOWN_GROUP_BORDER_POSITIONS = {
  top: 'top',
  bottom: 'bottom',
};
