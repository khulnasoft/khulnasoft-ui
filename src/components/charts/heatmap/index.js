import Heatmap from './heatmap.vue';

export default Heatmap;
