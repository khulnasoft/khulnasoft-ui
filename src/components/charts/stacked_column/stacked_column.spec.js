import { shallowMount } from '@vue/test-utils';

import { createMockChartInstance } from '~helpers/chart_stubs';
import { expectHeightAutoClasses } from '~helpers/chart_height';
import { LEGEND_LAYOUT_INLINE, LEGEND_LAYOUT_TABLE } from '~/utils/charts/constants';
import {
  mockDefaultStackedLineData,
  mockDefaultStackedBarData,
  mockSecondaryData,
} from '../../../utils/charts/mock_data';
import Chart from '../chart/chart.vue';
import ChartLegend from '../legend/legend.vue';
import ChartTooltip from '../tooltip/tooltip.vue';
import * as themeUtils from '../../../utils/charts/theme';
import StackedColumnChart from './stacked_column.vue';

let mockChartInstance;

jest.mock('echarts', () => ({
  getInstanceByDom: () => mockChartInstance,
}));

const defaultChartProps = {
  seriesNames: [],
  bars: mockDefaultStackedBarData,
  groupBy: ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'],
  xAxisType: 'category',
  xAxisTitle: 'January - December 2018',
  yAxisTitle: 'Commits',
};

describe('stacked column chart component', () => {
  let wrapper;

  const findChart = () => wrapper.findComponent(Chart);
  const findLegend = () => wrapper.findComponent(ChartLegend);
  const findDataTooltip = () => wrapper.findComponent(ChartTooltip);

  const emitChartCreated = () => findChart().vm.$emit('created', mockChartInstance);

  const createShallowWrapper = ({ props = {}, slots = {} } = {}) => {
    wrapper = shallowMount(StackedColumnChart, {
      propsData: { ...defaultChartProps, ...props },
      slots,
    });
    emitChartCreated();
  };

  beforeEach(() => {
    mockChartInstance = createMockChartInstance();
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  it('emits `created`, with the chart instance', () => {
    createShallowWrapper();

    return wrapper.vm.$nextTick(() => {
      expect(wrapper.emitted('created').length).toBe(1);
      expect(wrapper.emitted('created')[0][0]).toBe(mockChartInstance);
    });
  });

  it('should correctly render the chart', () => {
    createShallowWrapper();

    const chart = findChart();

    expect(chart.props('options')).toMatchSnapshot();
  });

  describe('with line data provided', () => {
    beforeEach(() => {
      createShallowWrapper({
        props: {
          bars: [],
          lines: mockDefaultStackedLineData,
        },
      });
    });
    it('should correctly render the chart', () => {
      expect(findChart().props('options')).toMatchSnapshot();
    });
  });

  describe('legend', () => {
    beforeEach(() => {
      mockChartInstance = {
        ...createMockChartInstance(),
        getOption: jest.fn().mockReturnValueOnce({
          series: [
            {
              type: 'solid',
              name: 'Foo Bar',
              data: [1, 2, 3, 4, 5],
            },
          ],
        }),
      };
    });

    it('is inline by default', () => {
      createShallowWrapper();

      return wrapper.vm.$nextTick(() => {
        expect(findLegend().props('layout')).toBe(LEGEND_LAYOUT_INLINE);
      });
    });

    it('is inline if correct prop value is set', () => {
      createShallowWrapper({ props: { legendLayout: LEGEND_LAYOUT_INLINE } });

      return wrapper.vm.$nextTick(() => {
        expect(findLegend().props('layout')).toBe(LEGEND_LAYOUT_INLINE);
      });
    });

    it('is tabular if correct prop value is set', () => {
      createShallowWrapper({ props: { legendLayout: LEGEND_LAYOUT_TABLE } });

      return wrapper.vm.$nextTick(() => {
        expect(findLegend().props('layout')).toBe(LEGEND_LAYOUT_TABLE);
      });
    });

    it('passes correct series info to legend', () => {
      createShallowWrapper();

      return wrapper.vm.$nextTick(() => {
        expect(findLegend().props('seriesInfo')).toEqual(
          expect.arrayContaining([
            {
              type: 'solid',
              name: 'Foo Bar',
              color: expect.any(String),
              data: [1, 2, 3, 4, 5],
              yAxisIndex: undefined,
            },
          ])
        );
      });
    });

    describe('when `includeLegendAvgMax` prop is disabled', () => {
      beforeEach(() => {
        createShallowWrapper({ props: { includeLegendAvgMax: false } });
      });

      it('passes correct series info to legend', () => {
        return wrapper.vm.$nextTick(() => {
          expect(findLegend().props('seriesInfo')).toEqual(
            expect.arrayContaining([
              {
                type: 'solid',
                name: 'Foo Bar',
                color: expect.any(String),
                data: undefined,
                yAxisIndex: undefined,
              },
            ])
          );
        });
      });
    });
  });

  describe('with secondary axis data provided', () => {
    const secondaryDataTitle = 'test secondary';

    beforeEach(() => {
      createShallowWrapper({
        props: {
          ...defaultChartProps,
          secondaryData: mockSecondaryData,
          secondaryDataTitle,
        },
      });
    });
    it('should correctly render the chart', () => {
      const chart = findChart();

      expect(chart.props('options')).toMatchSnapshot();
    });

    it('should set the secondary Y axis name', () => {
      const chart = findChart();

      expect(chart.props('options').yAxis[1].name).toEqual(secondaryDataTitle);
    });
  });

  describe('tooltip', () => {
    beforeEach(() => {
      createShallowWrapper();
    });

    it('is initialized', () => {
      expect(findDataTooltip().props('chart')).toBe(mockChartInstance);
    });
  });

  describe('color palette', () => {
    let paletteSpy;

    describe('default palette', () => {
      beforeEach(() => {
        paletteSpy = jest.spyOn(themeUtils, 'colorFromDefaultPalette');

        createShallowWrapper();
      });

      it('calls colorFromDefaultPalette', () => {
        expect(paletteSpy).toHaveBeenCalled();
      });
    });

    describe('custom palette', () => {
      beforeEach(() => {
        paletteSpy = jest.spyOn(themeUtils, 'colorFromDefaultPalette');

        createShallowWrapper({
          props: {
            customPalette: ['#FFFHHH', '#FFFJJJ', '#FFFIII', '#FFFKKK'],
          },
        });
      });

      it('does not call colorFromDefaultPalette', () => {
        expect(paletteSpy).not.toHaveBeenCalled();
      });

      it('matches the snapshot', () => {
        const chart = findChart();

        expect(chart.props('options')).toMatchSnapshot();
      });
    });
  });

  describe('height', () => {
    expectHeightAutoClasses({
      createComponent: (props) => createShallowWrapper({ props }),
      findContainer: () => wrapper,
      findChart,
    });
  });
});
