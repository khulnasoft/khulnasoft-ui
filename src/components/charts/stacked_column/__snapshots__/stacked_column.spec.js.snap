// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stacked column chart component color palette custom palette matches the snapshot 1`] = `
Object {
  "grid": Object {
    "bottom": 44,
    "left": 64,
    "right": 32,
    "top": 16,
  },
  "legend": Object {
    "show": false,
  },
  "series": Array [
    Object {
      "barMaxWidth": "50%",
      "data": Array [
        58,
        49,
        38,
        23,
        27,
        68,
        38,
        35,
        7,
        64,
        65,
        31,
      ],
      "emphasis": Object {
        "itemStyle": Object {
          "color": "rgba(255, undefined, undefined, 0.4)",
        },
      },
      "itemStyle": Object {
        "borderColor": "#FFFHHH",
        "borderWidth": 1,
        "color": "rgba(255, undefined, undefined, 0.2)",
      },
      "name": "Fun 1",
      "stack": Array [
        "Jan",
        "Feb",
        "Mar",
        "Apr",
        "May",
        "Jun",
        "Jul",
        "Aug",
        "Sep",
        "Oct",
        "Nov",
        "Dec",
      ],
      "type": "bar",
      "yAxisIndex": 0,
    },
    Object {
      "barMaxWidth": "50%",
      "data": Array [
        8,
        6,
        34,
        19,
        9,
        7,
        17,
        25,
        14,
        7,
        10,
        32,
      ],
      "emphasis": Object {
        "itemStyle": Object {
          "color": "rgba(255, undefined, undefined, 0.4)",
        },
      },
      "itemStyle": Object {
        "borderColor": "#FFFJJJ",
        "borderWidth": 1,
        "color": "rgba(255, undefined, undefined, 0.2)",
      },
      "name": "Fun 2",
      "stack": Array [
        "Jan",
        "Feb",
        "Mar",
        "Apr",
        "May",
        "Jun",
        "Jul",
        "Aug",
        "Sep",
        "Oct",
        "Nov",
        "Dec",
      ],
      "type": "bar",
      "yAxisIndex": 0,
    },
    Object {
      "barMaxWidth": "50%",
      "data": Array [
        67,
        60,
        66,
        32,
        61,
        54,
        13,
        50,
        16,
        11,
        47,
        28,
      ],
      "emphasis": Object {
        "itemStyle": Object {
          "color": "rgba(255, undefined, undefined, 0.4)",
        },
      },
      "itemStyle": Object {
        "borderColor": "#FFFIII",
        "borderWidth": 1,
        "color": "rgba(255, undefined, undefined, 0.2)",
      },
      "name": "Fun 3",
      "stack": Array [
        "Jan",
        "Feb",
        "Mar",
        "Apr",
        "May",
        "Jun",
        "Jul",
        "Aug",
        "Sep",
        "Oct",
        "Nov",
        "Dec",
      ],
      "type": "bar",
      "yAxisIndex": 0,
    },
    Object {
      "barMaxWidth": "50%",
      "data": Array [
        8,
        9,
        5,
        40,
        13,
        19,
        58,
        21,
        47,
        59,
        23,
        46,
      ],
      "emphasis": Object {
        "itemStyle": Object {
          "color": "rgba(255, undefined, undefined, 0.4)",
        },
      },
      "itemStyle": Object {
        "borderColor": "#FFFKKK",
        "borderWidth": 1,
        "color": "rgba(255, undefined, undefined, 0.2)",
      },
      "name": "Fun 4",
      "stack": Array [
        "Jan",
        "Feb",
        "Mar",
        "Apr",
        "May",
        "Jun",
        "Jul",
        "Aug",
        "Sep",
        "Oct",
        "Nov",
        "Dec",
      ],
      "type": "bar",
      "yAxisIndex": 0,
    },
  ],
  "xAxis": Object {
    "axisLabel": Object {
      "margin": 20,
      "verticalAlign": "bottom",
    },
    "axisLine": Object {
      "show": false,
    },
    "axisPointer": Object {
      "label": Object {
        "formatter": [Function],
      },
      "show": true,
      "type": "none",
    },
    "boundaryGap": true,
    "data": Array [
      "Jan",
      "Feb",
      "Mar",
      "Apr",
      "May",
      "Jun",
      "Jul",
      "Aug",
      "Sep",
      "Oct",
      "Nov",
      "Dec",
    ],
    "name": "January - December 2018",
    "nameLocation": "center",
    "splitLine": Object {
      "show": false,
    },
    "type": "category",
  },
  "yAxis": Array [
    Object {
      "axisLabel": Object {
        "formatter": [Function],
      },
      "axisTick": Object {
        "show": false,
      },
      "name": "Commits",
      "nameGap": 50,
      "nameLocation": "center",
      "type": "value",
    },
    Object {
      "axisLabel": Object {
        "formatter": [Function],
      },
      "axisTick": Object {
        "show": false,
      },
      "name": "",
      "nameGap": 50,
      "nameLocation": "center",
      "show": false,
      "type": "value",
    },
  ],
}
`;

exports[`stacked column chart component should correctly render the chart 1`] = `
Object {
  "grid": Object {
    "bottom": 44,
    "left": 64,
    "right": 32,
    "top": 16,
  },
  "legend": Object {
    "show": false,
  },
  "series": Array [
    Object {
      "barMaxWidth": "50%",
      "data": Array [
        58,
        49,
        38,
        23,
        27,
        68,
        38,
        35,
        7,
        64,
        65,
        31,
      ],
      "emphasis": Object {
        "itemStyle": Object {
          "color": "rgba(97, 122, 226, 0.4)",
        },
      },
      "itemStyle": Object {
        "borderColor": "#617ae2",
        "borderWidth": 1,
        "color": "rgba(97, 122, 226, 0.2)",
      },
      "name": "Fun 1",
      "stack": Array [
        "Jan",
        "Feb",
        "Mar",
        "Apr",
        "May",
        "Jun",
        "Jul",
        "Aug",
        "Sep",
        "Oct",
        "Nov",
        "Dec",
      ],
      "type": "bar",
      "yAxisIndex": 0,
    },
    Object {
      "barMaxWidth": "50%",
      "data": Array [
        8,
        6,
        34,
        19,
        9,
        7,
        17,
        25,
        14,
        7,
        10,
        32,
      ],
      "emphasis": Object {
        "itemStyle": Object {
          "color": "rgba(177, 79, 24, 0.4)",
        },
      },
      "itemStyle": Object {
        "borderColor": "#b14f18",
        "borderWidth": 1,
        "color": "rgba(177, 79, 24, 0.2)",
      },
      "name": "Fun 2",
      "stack": Array [
        "Jan",
        "Feb",
        "Mar",
        "Apr",
        "May",
        "Jun",
        "Jul",
        "Aug",
        "Sep",
        "Oct",
        "Nov",
        "Dec",
      ],
      "type": "bar",
      "yAxisIndex": 0,
    },
    Object {
      "barMaxWidth": "50%",
      "data": Array [
        67,
        60,
        66,
        32,
        61,
        54,
        13,
        50,
        16,
        11,
        47,
        28,
      ],
      "emphasis": Object {
        "itemStyle": Object {
          "color": "rgba(0, 144, 177, 0.4)",
        },
      },
      "itemStyle": Object {
        "borderColor": "#0090b1",
        "borderWidth": 1,
        "color": "rgba(0, 144, 177, 0.2)",
      },
      "name": "Fun 3",
      "stack": Array [
        "Jan",
        "Feb",
        "Mar",
        "Apr",
        "May",
        "Jun",
        "Jul",
        "Aug",
        "Sep",
        "Oct",
        "Nov",
        "Dec",
      ],
      "type": "bar",
      "yAxisIndex": 0,
    },
    Object {
      "barMaxWidth": "50%",
      "data": Array [
        8,
        9,
        5,
        40,
        13,
        19,
        58,
        21,
        47,
        59,
        23,
        46,
      ],
      "emphasis": Object {
        "itemStyle": Object {
          "color": "rgba(78, 127, 14, 0.4)",
        },
      },
      "itemStyle": Object {
        "borderColor": "#4e7f0e",
        "borderWidth": 1,
        "color": "rgba(78, 127, 14, 0.2)",
      },
      "name": "Fun 4",
      "stack": Array [
        "Jan",
        "Feb",
        "Mar",
        "Apr",
        "May",
        "Jun",
        "Jul",
        "Aug",
        "Sep",
        "Oct",
        "Nov",
        "Dec",
      ],
      "type": "bar",
      "yAxisIndex": 0,
    },
  ],
  "xAxis": Object {
    "axisLabel": Object {
      "margin": 20,
      "verticalAlign": "bottom",
    },
    "axisLine": Object {
      "show": false,
    },
    "axisPointer": Object {
      "label": Object {
        "formatter": [Function],
      },
      "show": true,
      "type": "none",
    },
    "boundaryGap": true,
    "data": Array [
      "Jan",
      "Feb",
      "Mar",
      "Apr",
      "May",
      "Jun",
      "Jul",
      "Aug",
      "Sep",
      "Oct",
      "Nov",
      "Dec",
    ],
    "name": "January - December 2018",
    "nameLocation": "center",
    "splitLine": Object {
      "show": false,
    },
    "type": "category",
  },
  "yAxis": Array [
    Object {
      "axisLabel": Object {
        "formatter": [Function],
      },
      "axisTick": Object {
        "show": false,
      },
      "name": "Commits",
      "nameGap": 50,
      "nameLocation": "center",
      "type": "value",
    },
    Object {
      "axisLabel": Object {
        "formatter": [Function],
      },
      "axisTick": Object {
        "show": false,
      },
      "name": "",
      "nameGap": 50,
      "nameLocation": "center",
      "show": false,
      "type": "value",
    },
  ],
}
`;

exports[`stacked column chart component with line data provided should correctly render the chart 1`] = `
Object {
  "grid": Object {
    "bottom": 44,
    "left": 64,
    "right": 32,
    "top": 16,
  },
  "legend": Object {
    "show": false,
  },
  "series": Array [
    Object {
      "data": Array [
        67,
        60,
        66,
        32,
        61,
        54,
        13,
        50,
        16,
        11,
        47,
        28,
      ],
      "itemStyle": Object {
        "color": "#617ae2",
      },
      "lineStyle": Object {
        "color": "#617ae2",
      },
      "name": "Stacked median line",
      "type": "line",
      "yAxisIndex": 0,
    },
  ],
  "xAxis": Object {
    "axisLabel": Object {
      "margin": 20,
      "verticalAlign": "bottom",
    },
    "axisLine": Object {
      "show": false,
    },
    "axisPointer": Object {
      "label": Object {
        "formatter": [Function],
      },
      "show": true,
      "type": "none",
    },
    "boundaryGap": true,
    "data": Array [
      "Jan",
      "Feb",
      "Mar",
      "Apr",
      "May",
      "Jun",
      "Jul",
      "Aug",
      "Sep",
      "Oct",
      "Nov",
      "Dec",
    ],
    "name": "January - December 2018",
    "nameLocation": "center",
    "splitLine": Object {
      "show": false,
    },
    "type": "category",
  },
  "yAxis": Array [
    Object {
      "axisLabel": Object {
        "formatter": [Function],
      },
      "axisTick": Object {
        "show": false,
      },
      "name": "Commits",
      "nameGap": 50,
      "nameLocation": "center",
      "type": "value",
    },
    Object {
      "axisLabel": Object {
        "formatter": [Function],
      },
      "axisTick": Object {
        "show": false,
      },
      "name": "",
      "nameGap": 50,
      "nameLocation": "center",
      "show": false,
      "type": "value",
    },
  ],
}
`;

exports[`stacked column chart component with secondary axis data provided should correctly render the chart 1`] = `
Object {
  "grid": Object {
    "bottom": 44,
    "left": 64,
    "right": 64,
    "top": 16,
  },
  "legend": Object {
    "show": false,
  },
  "series": Array [
    Object {
      "barMaxWidth": "50%",
      "data": Array [
        58,
        49,
        38,
        23,
        27,
        68,
        38,
        35,
        7,
        64,
        65,
        31,
      ],
      "emphasis": Object {
        "itemStyle": Object {
          "color": "rgba(97, 122, 226, 0.4)",
        },
      },
      "itemStyle": Object {
        "borderColor": "#617ae2",
        "borderWidth": 1,
        "color": "rgba(97, 122, 226, 0.2)",
      },
      "name": "Fun 1",
      "stack": Array [
        "Jan",
        "Feb",
        "Mar",
        "Apr",
        "May",
        "Jun",
        "Jul",
        "Aug",
        "Sep",
        "Oct",
        "Nov",
        "Dec",
      ],
      "type": "bar",
      "yAxisIndex": 0,
    },
    Object {
      "barMaxWidth": "50%",
      "data": Array [
        8,
        6,
        34,
        19,
        9,
        7,
        17,
        25,
        14,
        7,
        10,
        32,
      ],
      "emphasis": Object {
        "itemStyle": Object {
          "color": "rgba(177, 79, 24, 0.4)",
        },
      },
      "itemStyle": Object {
        "borderColor": "#b14f18",
        "borderWidth": 1,
        "color": "rgba(177, 79, 24, 0.2)",
      },
      "name": "Fun 2",
      "stack": Array [
        "Jan",
        "Feb",
        "Mar",
        "Apr",
        "May",
        "Jun",
        "Jul",
        "Aug",
        "Sep",
        "Oct",
        "Nov",
        "Dec",
      ],
      "type": "bar",
      "yAxisIndex": 0,
    },
    Object {
      "barMaxWidth": "50%",
      "data": Array [
        67,
        60,
        66,
        32,
        61,
        54,
        13,
        50,
        16,
        11,
        47,
        28,
      ],
      "emphasis": Object {
        "itemStyle": Object {
          "color": "rgba(0, 144, 177, 0.4)",
        },
      },
      "itemStyle": Object {
        "borderColor": "#0090b1",
        "borderWidth": 1,
        "color": "rgba(0, 144, 177, 0.2)",
      },
      "name": "Fun 3",
      "stack": Array [
        "Jan",
        "Feb",
        "Mar",
        "Apr",
        "May",
        "Jun",
        "Jul",
        "Aug",
        "Sep",
        "Oct",
        "Nov",
        "Dec",
      ],
      "type": "bar",
      "yAxisIndex": 0,
    },
    Object {
      "barMaxWidth": "50%",
      "data": Array [
        8,
        9,
        5,
        40,
        13,
        19,
        58,
        21,
        47,
        59,
        23,
        46,
      ],
      "emphasis": Object {
        "itemStyle": Object {
          "color": "rgba(78, 127, 14, 0.4)",
        },
      },
      "itemStyle": Object {
        "borderColor": "#4e7f0e",
        "borderWidth": 1,
        "color": "rgba(78, 127, 14, 0.2)",
      },
      "name": "Fun 4",
      "stack": Array [
        "Jan",
        "Feb",
        "Mar",
        "Apr",
        "May",
        "Jun",
        "Jul",
        "Aug",
        "Sep",
        "Oct",
        "Nov",
        "Dec",
      ],
      "type": "bar",
      "yAxisIndex": 0,
    },
    Object {
      "barMaxWidth": "50%",
      "data": Array [
        13,
        25,
        38,
        23,
        27,
        28,
        29,
        27,
        1,
        15,
        65,
        31,
      ],
      "emphasis": Object {
        "itemStyle": Object {
          "color": "rgba(207, 77, 129, 0.4)",
        },
      },
      "itemStyle": Object {
        "borderColor": "#cf4d81",
        "borderWidth": 1,
        "color": "rgba(207, 77, 129, 0.2)",
      },
      "name": "Secondary 1",
      "stack": "tiled",
      "type": "bar",
      "yAxisIndex": 1,
    },
  ],
  "xAxis": Object {
    "axisLabel": Object {
      "margin": 20,
      "verticalAlign": "bottom",
    },
    "axisLine": Object {
      "show": false,
    },
    "axisPointer": Object {
      "label": Object {
        "formatter": [Function],
      },
      "show": true,
      "type": "none",
    },
    "boundaryGap": true,
    "data": Array [
      "Jan",
      "Feb",
      "Mar",
      "Apr",
      "May",
      "Jun",
      "Jul",
      "Aug",
      "Sep",
      "Oct",
      "Nov",
      "Dec",
    ],
    "name": "January - December 2018",
    "nameLocation": "center",
    "splitLine": Object {
      "show": false,
    },
    "type": "category",
  },
  "yAxis": Array [
    Object {
      "axisLabel": Object {
        "formatter": [Function],
      },
      "axisTick": Object {
        "show": false,
      },
      "name": "Commits",
      "nameGap": 50,
      "nameLocation": "center",
      "type": "value",
    },
    Object {
      "axisLabel": Object {
        "formatter": [Function],
      },
      "axisTick": Object {
        "show": false,
      },
      "name": "test secondary",
      "nameGap": 50,
      "nameLocation": "center",
      "show": true,
      "type": "value",
    },
  ],
}
`;
